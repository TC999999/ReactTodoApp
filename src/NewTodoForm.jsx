import { useState } from "react";

const NewTodoForm = ({ addTodo }) => {
  const initialState = {
    todo: "",
  };

  const [formData, setFormData] = useState(initialState);
  const handleChange = (e) => {
    const { name, value } = e.target;
    setFormData((formData) => ({
      ...formData,
      [name]: value,
    }));
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    addTodo({ ...formData });
    setFormData(initialState);
  };

  return (
    <div id="todo-form-div">
      <form id="todo-form" onSubmit={handleSubmit}>
        <label htmlFor="todo">Next on the List</label>
        <input
          id="todo"
          type="text"
          name="todo"
          placeholder="Enter Task Here"
          value={formData.todo}
          onChange={handleChange}
        />

        <button>Add Todo</button>
      </form>
    </div>
  );
};

export default NewTodoForm;
