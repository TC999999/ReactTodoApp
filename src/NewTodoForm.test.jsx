import { render, fireEvent } from "@testing-library/react";
import NewTodoForm from "./NewTodoForm.jsx";
import { expect, vi } from "vitest";

it("renders without crashing", function () {
  render(<NewTodoForm />);
});

it("matches snapshot", function () {
  const { asFragment } = render(<NewTodoForm />);
  expect(asFragment()).toMatchSnapshot();
});

it("should be empty at the start", function () {
  const { getByLabelText } = render(<NewTodoForm />);
  const input = getByLabelText("Next on the List");
  expect(input).toBeEmptyDOMElement();
});

it("should have value when filled in", function () {
  const { getByLabelText } = render(<NewTodoForm />);
  const input = getByLabelText("Next on the List");

  fireEvent.change(input, { target: { value: "test-task" } });
  expect(input).toHaveValue("test-task");
});

it("should empty itself when submitted", function () {
  const addTodoMock = vi.fn();
  const { getByLabelText, queryByText } = render(
    <NewTodoForm addTodo={addTodoMock} />
  );
  expect(addTodoMock).not.toHaveBeenCalled();

  const input = getByLabelText("Next on the List");
  const btn = queryByText("Add Todo");

  fireEvent.change(input, { target: { value: "test-task" } });
  fireEvent.submit(btn);

  expect(addTodoMock).toHaveBeenCalled();
  expect(input).toBeEmptyDOMElement();
});
