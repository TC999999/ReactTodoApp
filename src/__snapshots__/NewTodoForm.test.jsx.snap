// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    id="todo-form-div"
  >
    <form
      id="todo-form"
    >
      <label
        for="todo"
      >
        Next on the List
      </label>
      <input
        id="todo"
        name="todo"
        placeholder="Enter Task Here"
        type="text"
        value=""
      />
      <button>
        Add Todo
      </button>
    </form>
  </div>
</DocumentFragment>
`;
